index.js

```diff
@@ -66,28 +70,28 @@ function register (server, options, next) {
   })
 
   // log when a request completes with an error
-  server.on('request-error', function (request, err) {
+  options.logEvents && options.logEvents.includes('request-error') && server.on('request-error', function (request, err) {
     request.logger.warn({
       res: request.raw.res,
       err: err
     }, 'request error')
   })
 
   // log when a request completes
-  server.on('response', function (request) {
+  options.logEvents && options.logEvents.includes('response') && server.on('response', function (request) {
     const info = request.info
     request.logger.info({
       res: request.raw.res,
       responseTime: info.responded - info.received
     }, 'request completed')
   })
 
-  server.ext('onPostStart', function (s, cb) {
+  options.logEvents && options.logEvents.includes('server-start') && server.ext('onPostStart', function (s, cb) {
     logger.info(server.info, 'server started')
     cb()
   })
 
-  server.ext('onPostStop', function (s, cb) {
+  options.logEvents && options.logEvents.includes('server-stop') && server.ext('onPostStop', function (s, cb) {
```

Comment:
this is not really nice to read. can you put it into an helper function, something like `extMaybe(server, options, 'onPostSto', function (s, cb) { })`